flink-connector-dynamodb/src/main/java/org/apache/flink/streaming/connectors/dynamodb/util/DynamoDbSerializationUtil.java

```diff
@@ -0,0 +1,195 @@
+                out.writeUTF(s);
+            }
+        } else if (value.hasBs()) {
+            out.writeByte(DynamoDbType.BINARY_SET.toByteValue());
+            out.writeInt(value.bs().size());
+            for (SdkBytes sdkBytes : value.bs()) {
+                byte[] bytes = sdkBytes.asByteArrayUnsafe();
+                out.writeInt(bytes.length);
+                out.write(bytes);
+            }
+        } else if (value.hasL()) {
+            out.writeByte(DynamoDbType.LIST.toByteValue());
+            List<AttributeValue> l = value.l();
+            out.writeInt(l.size());
+            for (AttributeValue attributeValue : l) {
+                serializeAttributeValue(attributeValue, out);
+            }
+        } else if (value.hasM()) {
+            out.writeByte(DynamoDbType.MAP.toByteValue());
+            Map<String, AttributeValue> m = value.m();
+            serializeItem(m, out);
+        } else {
+            throw new IllegalArgumentException("Attribute value must not be empty: " + value);
+        }
+    }
+
+    private static Map<String, AttributeValue> deserializeItem(DataInputStream in)
+            throws IOException {
+        int size = in.readInt();
+        Map<String, AttributeValue> item = new HashMap<>(size);
+        for (int i = 0; i < size; i++) {
+            String key = in.readUTF();
+            AttributeValue attributeValue = deserializeAttributeValue(in);
+            item.put(key, attributeValue);
+        }
+        return item;
+    }
+
+    private static AttributeValue deserializeAttributeValue(DataInputStream in) throws IOException {
+        int type = in.read();
+        DynamoDbType dynamoDbType = DynamoDbType.fromByteValue((byte) type);
+        return deserializeAttributeValue(dynamoDbType, in);
+    }
+
+    private static AttributeValue deserializeAttributeValue(
+            DynamoDbType dynamoDbType, DataInputStream in) throws IOException {
+        switch (dynamoDbType) {
+            case NULL:
+                return AttributeValue.builder().nul(true).build();
+            case STRING:
+                return AttributeValue.builder().s(in.readUTF()).build();
+            case NUMBER:
+                return AttributeValue.builder().n(in.readUTF()).build();
+            case BOOLEAN:
+                return AttributeValue.builder().bool(in.readBoolean()).build();
+            case BINARY:
+                int length = in.readInt();
+                byte[] bytes = new byte[length];
+                in.read(bytes);
+                return AttributeValue.builder().b(SdkBytes.fromByteArray(bytes)).build();
+            case STRING_SET:
+                int stringSetSize = in.readInt();
+                Set<String> stringSet = new LinkedHashSet<>(stringSetSize);
+                for (int i = 0; i < stringSetSize; i++) {
+                    stringSet.add(in.readUTF());
+                }
+                return AttributeValue.builder().ss(stringSet).build();
```

Comment:
Why use a `Set` here? Does this mean this is a non-symmetric transform? We could hit a situation where:
`!originalObject.equals(deserialize(serialize(originalObject)))`

If so, I think we should not be deduplicating here and the transform should be symmetrical